### PR TITLE
Bugfix/axa radio dom removal bug

### LINKED ISSUE #1752: AXA Radio Exception with Disconnected Callback
Issue reported by Yannick Streit from Zühlke Engineering:

"We’re using the AXA radio button component in version 2.1.5 for one of our features and are facing a strange issue. We would be glad if you can help us since it seems to be an issue with the AXA Radio component.
 
When we navigate forth an back in our wizard our React component containing the radio buttons gets mounted/unmounted accordingly. When navigating back to the component containing the radio buttons the “disconnectedCallback” of the AXA radio button gets called unexpectedly. Since the component is not set up yet, a “null pointer” exception is thrown by this component (see screenshot below from @axa-ch/radio/lib/index.react.js:751).
 
This is an excerpt the code where we use the radio buttons in a fieldset:
 
```
<div className="checkout-actions">
  <AXAFieldset horizontal>
       <AxaRadioButton
              name="service-offer-response"
             data-test-id="add-product"
             label={t('price.acceptOffer')}
             value="true"
             button
             noGap
             checked={props.selected === true}
             onChange={() => onToggle(true)}
       />
       <AxaRadioButton
             name="service-offer-response"
             data-test-id="remove-product"
             label={t('price.declineOffer')}
             value="false"
             button
             noGap
             checked={props.selected === false}
             onChange={() => onToggle(false)}
       />
  </AXAFieldset>
</div>
```
 
Could you help us out?"
Screenshot

### PR DESCRIPTION
**Pull request by markus-walther**
Fixes #1752.

### CODE REVIEW

File: src/components/10-atoms/radio/index.js [hunk 1]
```diff
@@ -244,11 +244,25 @@ class AXARadio extends NoShadowDOM {
...
+      return;
+    }
+    // clean up
     delete selectedRadioButton[name]; // help GC
     delete maxWidth[name]; // help GC
-    radioButtonGroup[name].delete(this.querySelector('.a-radio__content'));
-    if (radioButtonGroup[name].size === 0) {
-      delete radioButtonGroup[name]; // help GC
+    // one button less in the set
+    const ourButton = this.querySelector('.a-radio__content');
```
Comment (LucaMele) [inline src/components/10-atoms/radio/index.js]: We always separate css and JS Dom selection. please use `js-radio__content` instad of `a-radio__content`.

File: src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx [hunk 2]
```diff
@@ -60,32 +63,45 @@ const DemoRadiobuttonsControlled = () => {
...
         &nbsp;error&nbsp;
       </label>
+      <label htmlFor="visible">
+        <input
+          id="visible"
+          type="checkbox"
+          checked={!!visible}
+          onChange={handleVisible}
+        />
+        &nbsp;visible&nbsp;
```
Comment (raphaellueckl) [inline src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx:73]: Why the two whitespaces?

File: src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx [hunk 3]
```diff
@@ -60,32 +63,45 @@ const DemoRadiobuttonsControlled = () => {
         />
         &nbsp;error&nbsp;
       </label>
+      <label htmlFor="visible">
+        <input
+          id="visible"
+          type="checkbox"
+          checked={!!visible}
```
Comment (raphaellueckl) [inline src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx:70]: Why do you assign `visible` to `checked` instead of `checked`?

**Review by 01231 — DISMISSED** [review]
LGTM (approved when that Luca's comment is implemented)

File: src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx [hunk 4]
```diff
@@ -6,6 +6,7 @@ const DemoRadiobuttonsControlled = () => {
   const [button, setButton] = useState(true);
```
Comment (raphaellueckl) [inline src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx:6]: I don't really understand what more this demo-file is actually demoing now. Comments would be helpful.

File: src/components/10-atoms/radio/index.js (hunk 1 above)
Comment (markus-walther) [reply src/components/10-atoms/radio/index.js]: OK, although this was there already in the old code.

File: src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx (hunk 4 above)
Comment (markus-walther) [reply src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx:6]: // rendering axa-radios invisible cleanly removes them from DOM without exceptions being thrown

is the comment describing the 'what more'. But that one is in ui.test.js, so you're right I should add something also in DemoRadioButtonReact.jsx.

Comment (markus-walther) [reply src/components/10-atoms/radio/react/DemoRadioButtonReact.jsx:6]: Done

File: src/components/10-atoms/radio/index.js (hunk 1 above)
Comment (markus-walther) [reply src/components/10-atoms/radio/index.js]: Done

**Review by raphaellueckl — APPROVED** [review]
lgtm

**Pull request merged**